File: app/src/main/java/me/aluceps/sandbox/di/AppComponent.java
package me.aluceps.sandbox.di;

import javax.inject.Singleton;

import dagger.Component;
import me.aluceps.sandbox.MyApplication;

@Singleton
@Component(modules = {AppModule.class, HttpClientModule.class})
public interface AppComponent {

    void inject(MyApplication application);

    ActivityComponent plus(ActivityModule __);
}

File: app/src/main/java/me/aluceps/sandbox/di/AppComponent.kt
package me.aluceps.sandbox.di

import dagger.Component
import me.aluceps.sandbox.MyApplication
import javax.inject.Singleton

@Singleton
@Component(modules = [(AppModule::class), (HttpClientModule::class)])
interface AppComponent {

    fun plus(module: ActivityModule): ActivityComponent

    fun inject(application: MyApplication)
}
